test/integration/proposal-checks/IntegrationTestBadDebtFlows.sol

```diff
@@ -0,0 +1,371 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.13;
+
+import "@forge-std/Test.sol";
+
+import {PostProposalCheckFixture} from "@test/integration/proposal-checks/PostProposalCheckFixture.sol";
+
+contract IntegrationTestBadDebtFlows is PostProposalCheckFixture {
+    function testVoteForSDAIGauge() public {
+        uint256 mintAmount = governor.quorum(0);
+        /// setup
+        vm.prank(addresses.mainnet("TEAM_MULTISIG"));
+        rateLimitedGuildMinter.mint(address(this), mintAmount);
+        guild.delegate(address(this));
+
+        assertTrue(guild.isGauge(address(term)));
+        assertEq(guild.numGauges(), 1);
+        assertEq(guild.numLiveGauges(), 1);
+        assertEq(guild.balanceOf(address(this)), mintAmount);
+    }
+
+    function testAllocateGaugeToSDAI() public {
+        testVoteForSDAIGauge();
+
+        guild.incrementGauge(address(term), guild.balanceOf(address(this)));
+
+        assertEq(guild.totalWeight(), guild.balanceOf(address(this)));
+        assertTrue(guild.isUserGauge(address(this), address(term)));
+    }
+
+    function _supplyCollateralUserOne(
+        uint256 borrowAmount,
+        uint128 supplyAmount
+    ) public returns (bytes32 loanId) {
+        deal(address(sdai), userOne, supplyAmount);
+
+        uint256 startingCreditSupply = credit.totalSupply();
+
+        vm.startPrank(userOne);
+        sdai.approve(address(term), supplyAmount);
+        loanId = term.borrow(borrowAmount, supplyAmount);
```

Comment:
how is this working ? borrowAmount could be above supplyAmount * maxDebtPerCollateralToken